### PR TITLE
chore: update grading criteria and team formation text for the code 2 hackathon

### PR DESCRIPTION
**Pull request by serapieTuyishime**
# Submit a pull request

- [ ] This is not a duplicate of an existing pull request.
- [ ] No existing features have been broken without good reason.
- [ ] Your commit messages are detailed
- [ ] The code style guideline have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.
- [ ] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

Provide details about your pull request and what it adds, fixes, or changes.

:question:

## Breaking Changes

Describe what features are broken by this pull request and why, if any.

:question:

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

1. :question:

## Other Relevant Information

Provide any other important details below.

## Summary by CodeRabbit

- **New Features**
	- Updated the hackathon rubric display to show passing scores more clearly.
	- Enhanced team challenge setup by dynamically setting descriptions based on challenge details.
	- Introduced constants for grading criteria and team formation.

### CODE REVIEW

File: src/components/sections/challenges/SetupTeamChallenge.tsx [hunk 1]
```diff
@@ -20,13 +20,14 @@ interface Props {
 export default function SetupTeamChallenge(): JSX.Element {
   const { invite, challenge } = useMultiSelector<unknown, Props>({ invite: (state: IRootState) => state.invites.data, challenge: (state: IRootState) => state.challenges.current });
 
+  const description = challenge?.additionalInfo?.teamFormation || "Join DeArk - ICP.Hub Germany telegram group and find your teammates to complete the challenge with you."
```
Comment (Jonath-z) [inline src/components/sections/challenges/SetupTeamChallenge.tsx:23]: I would suggest to translate this sentense here

File: src/components/sections/challenges/SetupTeamChallenge.tsx [hunk 2]
```diff
@@ -20,13 +20,14 @@ interface Props {
...
   const { t } = useTranslation();
 
   return (
     <Section title="Submission">
       <div className="text-base font-normal text-slate-700 pt-8 pb-7 md:w-182.5">{t("communities.overview.challenge.team.setup.info")}</div>
       <div className="md:flex flex-row gap-5">
-        <FormTeamCard index={1} title="Form your team" description="Join DeArk - ICP.Hub Germany telegram group and find your teammates to complete the challenge with you." />
+        <FormTeamCard index={1} title="Form your team" description={description} />
```
Comment (Jonath-z) [inline src/components/sections/challenges/SetupTeamChallenge.tsx:30]: This title as well

File: src/components/sections/challenges/SetupTeamChallenge.tsx [hunk 3]
```diff
@@ -20,13 +20,14 @@ interface Props {
...
         {invite && !invite.team?.locked ? (
           <ConfirmTeamInvitation index={2} title="Submit your team" text={`The maximum team members for this challenge is ${challenge?.teamLimit || "3"} people`} invite={invite} />
```
Comment (Jonath-z) [inline src/components/sections/challenges/SetupTeamChallenge.tsx:32]: Here as well

**Review by Samantha-KY — APPROVED** [review]

File: src/types/course.d.ts [hunk 4]
```diff
@@ -78,6 +78,10 @@ export type Challenge = {
   teamLimit?: number;
   isHackathon?: boolean;
   multipleSubmissions?: boolean;
+  additionalInfo?: {
+    gradingCriteria?: string;
+    teamFormation?: string;
+  };
```
Comment (musayann) [inline src/types/course.d.ts:84]: I suggest the following type for additional information.

```ts
type Challenge = {
    // ... other challenge attributes
    additionInfos: {[type: string]: AdditionInfo};
}


type AdditionInfo = {
    title?: string;
    text: string;
}


const challenge = {
    additionInfos: {
        "GRADING_CRITERIA": {
            text: "The grading criteria is based on the following aspects: ..."
        },
        "TEAM_FORMATION": {
            text: "This is an example of how to solve the challenge."
        },
        "TIPS": {
            text: "Here are some tips to solve the challenge."
        }
    }
}
```

I would also store these additional info types as constants, mainly if used in multiple places.

eg:

```ts
// constants.ts

export const GRADING_CRITERIA = "GRADING_CRITERIA";
export const TEAM_FORMATION = "TEAM_FORMATION";
```

usage

```ts
const gradingInfo = challenge.additionalInfo[GRADING_CRITERIA];
```

**Review by musayann — CHANGES_REQUESTED** [review]

**Review by Jonath-z — APPROVED** [review]

**Review by musayann — APPROVED** [review]

**Pull request merged**